pkg/kubecost/summaryallocation_json_test.go

```diff
@@ -0,0 +1,129 @@
+package kubecost
+
+import (
+	"encoding/json"
+	"math"
+	"testing"
+	"time"
+
+	"github.com/opencost/opencost/pkg/util/timeutil"
+)
+
+func TestSummaryAllocationSetRangeResponse_MarshalJSON(t *testing.T) {
+	// Set a 1-day (start, end)
+	s := time.Date(2023, time.March, 13, 0, 0, 0, 0, time.UTC)
+	e := s.Add(timeutil.Day)
+
+	// Set some basic numbers that can be used to asset accuracy later
+	adjustment := -0.14
+	bytes := 2183471523842.00
+	cores := 2.78
+	cost := 12.50
+	gpus := 1.00
```

Comment:
Is it possible to have non-integer values for number of GPUs, cores, etc.?

Reply:
Certainly! I might only use half a CPU core over the span of an hour. Measurement is also imprecise, so things like averages may always report fractional values even if the idea of consuming half a RAM byte doesn't quite sound right. GPUs are particularly weird here because K8s supports "fractional" GPUs.

Reply:
Well when you explain it that way, it makes perfect sense! Thank you for taking the time to explain everything to me!